packages/amplify_datastore/ios/Classes/types/model/FlutterModelFieldType.swift

```diff
@@ -75,9 +76,38 @@ public struct FlutterModelFieldType {
                         fieldName: "ofModelName",
                         desiredType: "String")
                 }
-                return ModelFieldType.collection(of: ofModelName)
+                do {
+                    let embeddedType = try getPrimitiveTypeForEmbeddedCollection(typeName: ofModelName)
+                    return ModelFieldType.embeddedCollection(of: embeddedType, schema: nil)
+                } catch {
+                    return ModelFieldType.collection(of: ofModelName)
+                }
             default:
                 preconditionFailure("Could not create a ModelFieldType from \(fieldType)")
         }
     }
+
+    func getPrimitiveTypeForEmbeddedCollection(typeName: String) throws -> Codable.Type {
+        switch typeName {
+        case "int":
+            return Int.self
+        case "string":
+            return String.self
+        case "double":
+            return Double.self
+        case "bool":
+            return Bool.self
+        case "dateTime":
+            return Temporal.DateTime.self
+        case "Time":
```

Comment:
why is Time capitalized while other temporal types are not?

Reply:
it shouldn't be. Good catch